Modal

Deman4ig opened the pull request.

Deman4ig committed:
add solution for lesson-3 modal

Deman4ig committed:
add solution for lesson-3 modal

Deman4ig committed:
Merge branch 'modal' of github.com:Deman4ig/htmlcss-20210513_deman4ig into modal

jsru commented:
Добавляю преподавателя (@GSemikozov) для код-ревью.

GSemikozov commented on 05-lection3/02-modal/modal.css:
```diff
@@ -0,0 +1,62 @@
+body {
```
выносим за пределы файла стилей компонента)

Deman4ig replied:
сделано!

GSemikozov commented on 05-lection3/02-modal/modal.css:
```diff
@@ -0,0 +1,62 @@
+    left: 0;
+}
+
+.modal__wrapper {
```
есть ощущение что это лишний враппер и выравнивание модалки по середине можно перенести в .modal

Deman4ig replied:
не придумал ничего лучше чем поправить JS и выставлять display: flex вместо display: block. Соответственно justify-content и align-items перенес в .modal

GSemikozov commented on 05-lection3/02-modal/modal.css:
```diff
@@ -0,0 +1,62 @@
+    border-radius: 8px;
+    padding: 40px;
+    position: relative;
+    max-width: 50%;
```
думаю, не стоит таким образом ограничивать ширину и высоту - на мобилке это точно будет очень малюсенький блок))
лучше тогда как в макете - max-width в пикселях

Deman4ig replied:
ну если предположит что desktop-first, то модификаторы для мобилки можно добавить через @media:)

сделал через фиксированные px.

Deman4ig committed:
Refactor modal component

1. Moved generic styles out of component CSS
2. Replace percentage width and height of modal__body with values in
pixels
3. Removed modal__wrapper element

jsru commented:
Добавляю преподавателя (@GSemikozov) для код-ревью.

jsru commented:
Решение было обновлено, посмотрим что скажет @GSemikozov

Deman4ig committed:
Resolve conflict in sprites.svg file

jsru commented:
Решение было обновлено, посмотрим что скажет @GSemikozov

jsru commented:
Решение было обновлено, посмотрим что скажет @GSemikozov

jsru commented:
Решение было обновлено, посмотрим что скажет @GSemikozov
